### ISSUE TITLE
Install AC safety Switch

### ISSUE THREAD

**Issue by CCOSTAN** [CCOSTAN/Home-AssistantConfig#838]
https://amzn.to/3gntioj

cc: @jcallaghan - Do you think something like could be wired into a NodeMCU or something to send Alerts when Tripped?

**Comment by jcallaghan** [comment]
Yup, I'd use ESPHome on a D1 mini or other ESP based board. I'd then explore the ADC component and possibly the output component. You would want to pass a small current into one of those wires and then when your switch is open/off it will not draw any current which will be detected by the ADC sensor using the other part of the wire to form the circuit. 

Do you have the switch? Happy to jump on a Teams call later this week if you want some help.

**Comment by CCOSTAN** [comment]
I think I am going to hack a Wyze Sensor to leverage as a leak detector.. either secondary or primary.

**Comment by sreknob** [comment]
@jcallaghan @CCOSTAN - no need to get fancy with ADC for this. A simple GPIO binary sensor will do the job with ESPhome if you want to do it that way... Hook it up between ground and a pin. May need to be inverted.

```yaml
binary_sensor:
  - platform: gpio
    pin:
      number: D2
      mode: INPUT_PULLUP
    name: AC Leak Sensor
    device_class: moisture
```

Also, if you just want it to be a safety switch and not smart, you can just hook this up in series to your AC signaling wire to turn it off when water is detected. It's an intelligent but not smart solution :-)

**Comment by CCOSTAN** [comment]
Oh Wow.. that seems super easy as well.. But I think I am going to go the WYZE sensor route.. I have found a ton of videos that show just 2 solder points to remove the reed switch and replace with lead wires.  the biggest advantage is the 900 MHZ radio running from a button battery so no external power needed. 

Hmm.. I might have to see if there is an actual outlet up there.. if there is, I might change my mind back again and do the NODENCU/ESPHome route since I could set it and forget it Forever. :)  Plus add more sensors to the board as well.. 

For the GPIO sensor, I could just use a bread board and have 2 wires coming out of ground and pin right? The water would provide the 'close'.

I also think I'm too scared to hook anything in series with the AC switch.  I am too naive when it comes to voltage etc.. My plan is to take the 'smart' signal from the Wyze or NODEMCU and then have HA just turn off the NEST thermostats.   No actual power cuts.

**Comment by sreknob** [comment]
> Oh Wow.. that seems super easy as well.. But I think I am going to go the WYZE sensor route.. I have found a ton of videos that show just 2 solder points to remove the reed switch and replace with lead wires. the biggest advantage is the 900 MHZ radio running from a button battery so no external power needed.

Sounds like an excellent "wire free" solution.

>For the GPIO sensor, I could just use a bread board and have 2 wires coming out of ground and pin right? The water would provide the 'close'.

Yes, just hook the pins up and you should be good. Whether you "invert" the sensor depends on whether the switch is NO (normally open) or NC (normally closed). For the switch you link, it appears to be NC.

> I also think I'm too scared to hook anything in series with the AC switch. I am too naive when it comes to voltage etc.. My plan is to take the 'smart' signal from the Wyze or NODEMCU and then have HA just turn off the NEST thermostats. No actual power cuts.

For this, it works best if it's just a signaling wire (low voltage) but that only applied to central-air type units that hook up to a furnace, so not useful if you unit is separate.

**Comment by CCOSTAN** [comment]
Installed Normal T safety Switch on Main Unit. 

https://amzn.to/2XsOpku

Once the switch is triggered, it cuts power and Nest alerts me that the unit went offline for investigation.  Seems like a reasonable solution. 

Will CLOSE once I have the T Switch on the secondary unit as well.

**Comment by CCOSTAN** [comment]
September 14, 2026 update: the upstairs air-handler safety float was replaced, but it still did not shut the unit down. We then tested with a known-good float switch from the main air conditioner and the upstairs unit still did not shut down. That rules out the float itself and points to a wiring/control-path issue. Leaving this open for wiring investigation or the next HVAC visit.
